docs/admin/ha-master-gce.md

```diff
@@ -0,0 +1,159 @@
+
+* `MULTIZONE=true` - to prevent removal of master replicas kubelets from zones different than server's default zone.
+Required if you want to run master replicas in different zones, which is recommended.
+
+* `ENABLE_ETCD_QUORUM_READS=true` - to ensure that reads from all API servers will return most up-to-date data.
+If true, reads will be directed to leader etcd replica.
+Setting this value to true is optional: reads will be more reliable but will also be slower.
+
+In addition, we may specify in which GCE zone the first master replica will be created by setting:
+
+* `KUBE_GCE_ZONE=zone` - zone where the first master replica will run.
+
+The sample command to set up the HA-compatible cluster:
+
+```
+$ MULTIZONE=true KUBE_GCE_ZONE=europe-west1-b  ENABLE_ETCD_QUORUM_READS=true ./cluster/kube-up.sh
+```
+
+Please note that execution of the comments above will create a cluster with one master,
+but the cluster will allow adding a new master replicas in future.
+
+### Adding a new master replica
+
+After creation of HA-compatible cluster, we should add some master replicas to it.
+Creation of a master replica is also done by kube-up script with the following flags:
+
+* `KUBE_REPLICATE_EXISTING_MASTER=true` - to create a replica of an existing
+master.
+
+* `KUBE_GCE_ZONE=zone` - zone where the master replica will run.
+Should be in the same region as other replicas' zones.
+
+* you don't need to set `MULTIZONE` or `ENABLE_ETCD_QUORUM_READS` flags as they values will be inherited from already running clusters
+(we assume that the flag were set during starting HA-compatible cluster).
+
+The sample command:
+
+```
+$ KUBE_GCE_ZONE=europe-west1-c KUBE_REPLICATE_EXISTING_MASTER=true ./cluster/kube-up.sh
+```
+
+### Removing master replica
+
+A master replica may be removed using kube-down script with the following flags:
+
+* `KUBE_DELETE_NODES=false` - to restrain deletion of kubelets.
+
+* `KUBE_GCE_ZONE=zone` - the zone from where master replica will be removed.
+ 
+* `KUBE_REPLICA_NAME=replica_name` - (optional) the name of master replica to remove.
+If empty: any replica from the given zone will be removed.
+
+The sample command:
+
+```
+$ KUBE_DELETE_NODES=false KUBE_GCE_ZONE=europe-west1-c ./cluster/kube-down.sh
+```
+
+### In case of replica failure
+
+If one of master replica in cluster is broken, we should remove it and add a
+new replica in the same zone. The sample commands:
+
+1. Remove the broken replica:
+
+```
+$ KUBE_DELETE_NODES=false KUBE_GCE_ZONE=replica_zone KUBE_REPLICA_NAME=replica_name ./cluster/kube-down.sh
+```
+
+2. Add a new replica in place of the old one:
+
+```
+$ KUBE_GCE_ZONE=replica-zone KUBE_REPLICATE_EXISTING_MASTER=true ./cluster/kube-up.sh
+```
+
+### Deployment best practices
+
+* Try to place master replicas in different zones. During a zone failure, all master placed inside the zone will fail.
+o prevent zone failure, also place nodes in  multiple zones
```

Comment:
'prevent' is incorrect. This just reduces the impact on your cluster, it doesn't prevent a zone from failing.